sdk/core/core/inc/az_span_builder.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_SPAN_BUILDER_H
+#define AZ_SPAN_BUILDER_H
+
+#include <az_span.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct {
+  az_span span;
+  size_t i;
```

Comment:
Can you add some comments.  The size on the builder is unnecessary.  I expect the size of the span in the builder to be the same as the size of the span it contains.  Why track the size twice?

Reply:
@gilbertw I've added some comments. The builder has a buffer (previously known as a `.span`) and a position (previously known as `.i`). The buffer (the type is a `az_span`) has a size which is capacity. `.position` is a current write position and should never be bigger than `.buffer.size`. We may rename it to `.size` I don't care much.